### PR TITLE
Reorder Sample list columns

### PR DESCRIPTION
**Pull request by devduarte**
-Related #1362

## Before this fix
Screen Shot 2021-11-12 at 14 00 00

## After this fix
Screen Shot 2021-11-12 at 14 01 06

### CODE REVIEW

File: app/models/concerns/specimen_role.rb [hunk 1]
```diff
@@ -7,6 +7,11 @@ module SpecimenRole
                            allow_blank: true,
                            message: "is not within valid options"
 
+    def specimen_role_description(id)
+      specimen_role_id = id || ""
+      self.class.specimen_role_description(specimen_role_id)
+    end
```
Comment (ysbaddaden) [inline app/models/concerns/specimen_role.rb:13]:
```suggestion
    def specimen_role_description
      self.class.specimen_role_description(specimen_role)
    end
```

We don't need to pass the specimen role id to the method, we already have access to the `#specimen_role` attribute. Also the class method will already check if the value is nil.

This allows us to use `sample.specimen_role_description` and `batch.specimen_role_description` directly :raised_hands:

File: app/models/concerns/specimen_role.rb [hunk 2]
```diff
@@ -18,9 +23,14 @@ def specimen_role_descriptions
       YAML.load_file(File.join("app", "models", "config", "specimen_roles.yml"))["specimen_roles"]
     end
 
+    def specimen_role_description(id)
+      specimen_role_id = id || ""
+      "#{specimen_role_id.upcase} - #{specimen_role_descriptions[specimen_role_id] || ""}"
+    end
```
Comment (ysbaddaden) [inline app/models/concerns/specimen_role.rb:29]: I understand that the `specimen_role` attribute is optional? Then maybe we should return nil or an empty string in that case? For example:

```suggestion
    def specimen_role_description(id)
      "#{id.upcase} - #{specimen_role_descriptions[id]}" if id
    end
```

**Review by ysbaddaden — APPROVED** [review]
Just a few suggestions to make the feature even nicer :)

File: app/models/concerns/specimen_role.rb (hunk 1 above)
Comment (ftarulla) [reply app/models/concerns/specimen_role.rb:13]: Also keep in mind that changes were made to this concern in #1366 (that should be merged before this PR)

**Review by ysbaddaden — APPROVED** [review]
LGTM!

**Pull request merged**